### PR TITLE
[FSDE-3601] Add support of reordering locked columns in Column customization submodule

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://27e9ad657872a87fee520a247e56799536c1afc1-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ee17054ed998c8af76e7f82f392bc61e604f4aa4-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://21f1e0910eb364acc6d7c2b8cac1a77a4ce0809a-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://7f7ee309507c6a6c2efea7f30d0dd198f2573e61-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/5/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://00186918888b28e177803eedaf559d6a6ce365e4-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/6/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/7/. :x:

File: src/chi-vue/src/components/column-customization/SelectedColumns.tsx [hunk 1]
```diff
@@ -37,15 +41,31 @@ export default class ColumnCustomizationSelectedColumns extends Vue {
...
+    if (selectButton && deselectButton) {
+      if (column.locked) {
+        selectButton.disabled = true;
+        deselectButton.disabled = true;
+        return;
+      }
+    }
+
+    selectButton.disabled = false;
+    deselectButton.disabled = false;
```
Comment (janibolkvadze) [inline src/chi-vue/src/components/column-customization/SelectedColumns.tsx:59]: Here you are selecting the element and changing their state. Instead, when generating these components, you need to set
disabled = {CONDITION_OF_DISABLING}
and when selecting options in selected columns list, change the CONDITION_OF_DISABLING so that the button renders in the proper state.

File: src/chi-vue/src/components/column-customization/AvailableColumns.tsx [hunk 2]
```diff
@@ -25,9 +25,23 @@ export default class ColumnCustomizationAvailableColumns extends Vue {
     }
   }
 
+  changeSelectRowsState() {
+    const selectButton = document.querySelector('.icon-chevron-right')?.parentElement
```
Comment (janibolkvadze) [inline src/chi-vue/src/components/column-customization/AvailableColumns.tsx:29]: Why are these necessary here? Please, find my comment below regarding CONDITION_OF_DISABLING

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://1807086c334f23bde111c72ce93c0c3207a8a645-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/components/column-customization/SelectedColumns.tsx (hunk 1 above)
Comment (josecarlosgonzalezv) [reply src/chi-vue/src/components/column-customization/SelectedColumns.tsx]: Solved.

File: src/chi-vue/src/components/column-customization/AvailableColumns.tsx (hunk 2 above)
Comment (josecarlosgonzalezv) [reply src/chi-vue/src/components/column-customization/AvailableColumns.tsx]: Solved.

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/9/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ad15589db0650537dff452c5d39b522c913a3134-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/10/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://b4f787438d91bca5b3995562082cd807fe091808-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/11/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://0976938733457ee3f6e8ab2f9e4ef029e068cc64-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/12/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://4f723091d7c5bbe0dfff886508e1bf6985fd0025-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/13/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://cca73874e7b211ad2d5d4c1ef1dbec450fbb2aba-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/14/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://cf614fcd906671460d4bff6c615a777eccdf61d1-chi-pr.enterprise.ctl.io

**Review by SergioQCL — APPROVED** [review]
LGTM

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx [hunk 3]
```diff
@@ -172,6 +167,34 @@ export default class ColumnCustomizationContent extends Vue {
...
+        this._selectedStandardColumns?.forEach((standardColumn: DataTableColumn) => {
+          if (columnSelected === standardColumn.name) {
+            filterSelectedColumns.push(standardColumn);
+          }
+        });
+      });
+    }
+
+    this._isSelectedColumnLocked = filterSelectedColumns.some((column: DataTableColumn) => column.locked);
+    this.$forceUpdate();
```
Comment (aryaroudi) [inline src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:191]: I don't recommend using this. Can you re-think your solution and implement it in a way to not use forceupdate?

**Review by aryaroudi — CHANGES_REQUESTED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/15/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c6894b5e64084509fa69f92fa420be69f02af3fd-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/16/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://a30f6f178932203bd9823d66e3634725b17d538e-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/17/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://3577d26f28b1c83611f5fed788c419fd9c2f15e4-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/18/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://51b2f1a0b8d2f158fd26a6ffbbb7aabda3c257ba-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx (hunk 3 above)
Comment (dani-cl-madrid) [reply src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:191]: I agree with @aryaroudi , forceUpdate should be avoided. @josecarlosgonzalezv

File: src/chi-vue/src/components/column-customization/AvailableColumns.tsx [hunk 4]
```diff
@@ -25,9 +26,21 @@ export default class ColumnCustomizationAvailableColumns extends Vue {
     }
   }
 
+  _triggerSelectRowState(column: DataTableColumn) {
+    this.$props.handleSelectColumn(column);
```
[the remark this answers is not in the document]
Comment (josecarlosgonzalezv) [reply src/chi-vue/src/components/column-customization/AvailableColumns.tsx]: Solved this.

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx (hunk 3 above)
Comment (josecarlosgonzalezv) [reply src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx]: Yes, I am agree too after reading some docs about it. 

After trying some different things, I decided to use a ref to manipulate the disabled attribute, you can check it now.

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/19/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://f067023a7efcdcfd2a3970e0261f73d70031509a-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/20/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://7d90befb393c35a5aeb29fcbf7f4c987f66fdcdf-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/components/column-customization/AvailableColumns.tsx [hunk 5]
```diff
@@ -7,6 +7,7 @@ import { UTILITY_CLASSES } from '@/constants/classes';
 @Component
 export default class ColumnCustomizationAvailableColumns extends Vue {
   @Prop() availableColumns?: DataTableColumn[];
+  @Prop() handleSelectColumn?: (column: DataTableColumn) => void;
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/AvailableColumns.tsx:10]: Instead of receiving a handler, this component should emit an event, and the parent component should listen to that event and handle the column selection. That is the common front-end applications architecture.

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx [hunk 6]
```diff
@@ -11,19 +11,17 @@ import { _changeOrder } from '@/components/column-customization/utils';
...
-  @Prop() selectedLockedColumns?: DataTableColumn[];
   @Prop() selectedStandardColumns?: DataTableColumn[];
 
   key = 0;
   _availableColumns?: DataTableColumn[] = [];
-  _selectedLockedColumns?: DataTableColumn[] = [];
   _selectedStandardColumns?: DataTableColumn[] = [];
   _availableColumnsComponent?: AvailableColumns;
   _selectedColumnsComponent?: SelectedColumns;
+  _isSelectedColumnLocked = false;
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:21]: This name seems to be related to a *single* locked column. Is it be possible to have more than 1 selected locked columns at the same time. If so, this name is not good.

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx [hunk 7]
```diff
@@ -172,6 +173,30 @@ export default class ColumnCustomizationContent extends Vue {
...
+      columnsSelected.forEach((columnSelected: string) => {
+        this._selectedStandardColumns?.forEach((standardColumn: DataTableColumn) => {
+          if (columnSelected === standardColumn.name) {
+            filterSelectedColumns.push(standardColumn);
+          }
+        });
+      });
+    }
+
+    this._isSelectedColumnLocked = filterSelectedColumns.some((column: DataTableColumn) => column.locked);
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:196]: It seems no locked columns can exist in the filterSelectedColumns. It comes from columnsSelected, which comes from _selectedStandardColumns that does not have locked columns.

File: src/chi-vue/src/components/column-customization/SelectedColumns.tsx [hunk 8]
```diff
@@ -7,8 +7,8 @@ import { UTILITY_CLASSES } from '@/constants/classes';
 
 @Component
 export default class ColumnCustomizationSelectedColumns extends Vue {
-  @Prop() lockedColumns?: DataTableColumn[];
   @Prop() standardColumns?: DataTableColumn[];
+  @Prop() handleSelectColumn?: (column: DataTableColumn) => void;
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/SelectedColumns.tsx:11]: Again, this is not a good practice.

File: src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx [hunk 9]
```diff
@@ -191,8 +219,8 @@ export default class ColumnCustomizationContent extends Vue {
         </div>
         <div class={UTILITY_CLASSES.FLEX.FLEX_GROW1}>
           <SelectedColumns
-            locked-columns={this._selectedLockedColumns}
             standard-columns={this._selectedStandardColumns}
+            handle-select-column={this._handleSelectColumn}
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/ColumnCustomizationModalContent.tsx:223]: Remember to not pass function handlers as parameters. Emit events from the descendant component instead and set your handler in an eventListener.

**Review by dani-cl-madrid — CHANGES_REQUESTED** [review]

File: src/chi-vue/src/components/column-customization/ColumnCustomization.tsx [hunk 10]
```diff
@@ -24,7 +24,6 @@ export default class ColumnCustomization extends Vue {
   key = 0;
   _chiModal: any;
   _availableColumns?: DataTableColumn[] = [];
-  _selectedLockedColumns?: DataTableColumn[] = [];
   _selectedStandardColumns?: DataTableColumn[] = [];
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/column-customization/ColumnCustomization.tsx:27]:
```suggestion
  _selectedColumns?: DataTableColumn[] = [];
```

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/21/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ffa031bbb146f923b8eca42298b647a3dfff1b86-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/22/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://16b02495863ad0c631dc251c16db35b0a4ee7422-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/23/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://4b929b9a10a8fc7b5e25cb70f6c41f2a6c8ed5a2-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/24/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ce04ee555df0d39bd0f07d5019698e2592c28151-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/25/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://0db7875b6b08ed5c51b1826bfdc3d086a8dea925-chi-pr.enterprise.ctl.io

**Review by aryaroudi — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/26/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://1304f2e558d2abed584c2f14096bd03ad233f21c-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/27/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://36f91bdd7297e6f039c10aa20befc9f3c513d676-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://a293b51fa404181084a4bae01db3ab8f5824364c-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/29/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://51bb3cb908531c8f6a212081d6dbd124e690348a-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-158/30/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://bf0fef30caa7da15ea061d00171e9b6297bd52b3-chi-pr.enterprise.ctl.io

**Review by dani-cl-madrid — APPROVED** [review]

**Review by jllr — APPROVED** [review]
